internal/v1/api.yaml

```diff
@@ -1461,7 +1463,27 @@ components:
           description: |
             Determines whether a valid subscription manager (candlepin) identity is required to
             access this repository. Consumer certificates will be used as client certificates when
-            fetching metadata and content.
+             fetching metadata and content.
+    AAPRegistration:
+      type: object
+      x-go-name: AAPRegistration
+      additionalProperties: false
+      required:
+        - ansible_controller_url
+        - job_template_id
+        - host_config_key
+      properties:
+        ansible_controller_url:
+          type: string
+          example: "example.towerhost.net"
+        job_template_id:
+          type: integer
+          example: 38
+        host_config_key:
+          type: string
+          example: "44d7507f2ead49af5fca80aa18fd24bc"
+        tls_certificate_authority:
+          type: string
```

Comment:
Nitpick: this could have `x-go-type-skip-optional-pointer` and then it will be just string, when empty it will be omitted as `omitempty` is added by default by the generator.